pkgs/desktops/cinnamon/cinnamon-session/default.nix

```diff
@@ -0,0 +1,51 @@
+{ cinnamon-desktop, cinnamon-settings-daemon, dbus-glib, docbook_xsl, docbook_xml_dtd_412, fetchFromGitHub, glib, gsettings_desktop_schemas, gtk3, libcanberra, libxslt, makeWrapper, meson, ninja, pkgconfig, python3, stdenv, systemd, wrapGAppsHook, xapps, xmlto, xorg, gnome2, cmake, libexecinfo }:
+
+stdenv.mkDerivation rec {
+  pname = "cinnamon-session";
+  version = "4.2.1";
+
+  src = fetchFromGitHub {
+    owner = "linuxmint";
+    repo = pname;
+    rev = version;
+    sha256 = "1jrwjnrcmp5m9vlp42ql79bxic5nrs37kkgcvgyhvvvskvdwpyfw";
+  };
+
+  /* Run-time dependency libelogind found: NO (tried pkgconfig and cmake)
+  Run-time dependency libsystemd-login found: NO (tried pkgconfig and cmake)
+  Run-time dependency libsystemd found: NO (tried pkgconfig and cmake) */
+
+  buildInputs = [ cinnamon-desktop cinnamon-settings-daemon dbus-glib docbook_xsl docbook_xml_dtd_412 glib gsettings_desktop_schemas gtk3 libcanberra libxslt makeWrapper pkgconfig python3 xapps xmlto xorg.xtrans xorg.libXtst gnome2.GConf systemd ];
+  nativeBuildInputs = [ meson ninja wrapGAppsHook cmake libexecinfo ];
+
+  NIX_CFLAGS_COMPILE = "-I${glib.dev}/include/gio-unix-2.0";
+  configureFlags = [ "--enable-systemd" ];
+
+  postPatch = ''
+    chmod +x data/meson_install_schemas.py # patchShebangs requires executable file
+    patchShebangs data/meson_install_schemas.py
+
+    substituteInPlace cinnamon-session/meson.build \
+      --replace "# elogind" "dbus_glib"
```

Comment:
Apparently they're completely ported to gdbus https://github.com/linuxmint/cinnamon-session/commit/16371b437e7bb98610bca2c814d43491a5d65173, but I think there's one they missed https://github.com/linuxmint/cinnamon-session/blob/master/cinnamon-session/csm-util.c#L35.